apis/io-engine/protobuf/v1/snapshot-rebuild.proto

```diff
@@ -29,10 +41,12 @@ message CreateSnapshotRebuildRequest {
   bool                             resume = 3;
   // A bitmap where set bits indicate which
   // segment of the snapshot are allocated and therefore
-  // need to be transferred for the rebuild.
+  // need to be transferred from the source to rebuild.
   // If not specified the replica shall retrieve it
   // using nvme-admin command.
   optional SnapshotRebuildBitmap   bitmap = 4;
+  /// When running on the snapshot node, use bitmap mode as well.
+  bool                       local_bitmap = 8;
```

Comment:
Do we really have to provide a bool switch to use or not-use bitmap? Shall we not simply use bitmap internally if it is present?

Reply:
This one is a bit awkward, perhaps naming is not entirely correct.
Basically we can either receive the bitmap via gRPC and walk it, or we can retrieve the bitmap and walk, or we can just walk the device.
Maybe this should be an oneof instead,
oneof {
SnapshotRebuildBitmap.
FetchBitmap
NoBitmap,
} 
Suggestions?

Reply:
> receive the bitmap via gRPC and walk it

Receive via gRPC from where? - Considering a case say source snapshot is of a replica on node1, and replica rebuild destination is on node2.

Reply:
Bitmap can be specified via gRPC parameters, L47

Reply:
ah right. `oneof` would be nice. Something like...

```
{
bitmap_consumer_remote,
bitmap_consumer_local,
no_bitmap
}
```

Reply:
hmm doesn't didn't quite work here as it expects each one to be a field, so we'd have to add fields, ex:
```
  oneof RebuildBitmapArgs {
    int NoBitmap = 1;
    SnapshotRebuildBitmap SpecifiedArgs = 2;
    int NvmeAdmin = 3;
  }
```

Just the type SnapshotRebuildBitmap with a bool is probably simpler:
```
  // A bitmap where set bits indicate which
  // segment of the snapshot are allocated and therefore
  // need to be transferred from the source to rebuild.
  // If not specified the replica shall retrieve it
  // using nvme-admin command.
  optional SnapshotRebuildBitmap   bitmap = 4;
  // Use the bitmap for the rebuild, or just walk the device range.
  // If the bitmap is specified via the args use it, otherwise fetch
  // it from the snapshot.
  bool                         use_bitmap = 8;
```

Reply:
ok, let's keep it as is then.